spring-core/src/main/java/org/springframework/util/StringUtils.java

```diff
@@ -810,8 +811,9 @@ public static TimeZone parseTimeZoneString(String timeZoneString) {
 		}
 		List<String> result = new ArrayList<>();
 		result.addAll(Arrays.asList(array1));
+		Set<String> stringSet = new HashSet<>(result);
```

Comment:
You can create HashSet from Arrays.asList(array1) (the "result" list is not required). 

But this way is not clear. I see an easier way:
/*
Set<String> result = new HashSet<>();
Collections.addAll(result, array1);
Collections.addAll(result, array2);
*/